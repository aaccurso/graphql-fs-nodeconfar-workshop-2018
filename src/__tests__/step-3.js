const { graphql } = require('graphql');
const schema = require('../schema');

describe('step 3', () => {
  it('should query files and dirs recursively', async () => {
    const source = `
      query filesAndDirs {
        dirs {
          name
          type
          files {
            name
            type
          }
          dirs {
            name
            type
          }
        }
      }
    `;

    expect(await graphql({ schema, source })).toMatchSnapshot();
  });
});
