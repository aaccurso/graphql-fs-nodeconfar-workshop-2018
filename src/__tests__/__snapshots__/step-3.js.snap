// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`step 3 should query files and dirs recursively 1`] = `
Object {
  "data": Object {
    "dirs": Array [
      Object {
        "dirs": Array [
          Object {
            "name": "Father",
            "type": "Dir",
          },
          Object {
            "name": "Mother",
            "type": "Dir",
          },
        ],
        "files": Array [
          Object {
            "name": "Homer_Simpson.png",
            "type": "File",
          },
        ],
        "name": "Father",
        "type": "Dir",
      },
      Object {
        "dirs": Array [
          Object {
            "name": "Father",
            "type": "Dir",
          },
          Object {
            "name": "Mother",
            "type": "Dir",
          },
        ],
        "files": Array [
          Object {
            "name": "Marge_Simpson.png",
            "type": "File",
          },
          Object {
            "name": "Patty_Bouvier.png",
            "type": "File",
          },
          Object {
            "name": "Selma_Bouvier.png",
            "type": "File",
          },
        ],
        "name": "Mother",
        "type": "Dir",
      },
    ],
  },
}
`;
